sycl/plugins/level_zero/pi_level_zero.hpp

```diff
@@ -268,8 +272,9 @@ struct _pi_context : _pi_object {
 
 struct _pi_queue : _pi_object {
   _pi_queue(ze_command_queue_handle_t Queue, pi_context Context,
-            pi_device Device)
-      : ZeCommandQueue{Queue}, Context{Context}, Device{Device} {}
+            pi_device Device, pi_uint32 QueueBatchSize)
+      : ZeCommandQueue{Queue}, Context{Context}, Device{Device},
+        QueueBatchSize{QueueBatchSize} {}
```

Comment:
why do we need batch size to be a member of a queue instead of just using `ZeCommandListBatchSize`?

Reply:
I was setting this code up for the future.  By making this a field of the queue that gets set at queue creation time based on the global ZeCommandListBatchSize global variable/env var, I can allow a heuristic to change this on a queue specific basis without introducing any race/thread unsafe global var uses since fields of the queue are now all guarded in a thread safe manner. So, while not specifically necessary, I think it is an excellent way to make it easier to add heuristics for batching in the future.

Reply:
OK, but please clarify this intention of `QueueBatchSize` in a comment.

Reply:
Will do.